src/core_atmosphere/Registry.xml

```diff
@@ -336,6 +336,18 @@
                      possible_values="Non-negative real values"/>
 	</nml_record>
 
+        <nml_record name="self_monitoring" in_defaults="true">
+                <nml_option name="config_max_median_time" type="real" default_value="1.0"
+                     units="s"
+                     description="Maximum allowable median time for a timestep, beyond which the model will terminate"
+                     possible_values="Positive real values"/>
+
+                <nml_option name="config_median_sample_size" type="integer" default_value="100"
+                     units="-"
+                     description="Number of timesteps that are sampled before computing median timestep time"
+                     possible_values="Positive integer values"/>
+        </nml_record>
```

Comment:
These `self_monitoring` namelist options were included to catch real-time HWT simulations that were progressing too slowly, and are not part of the changes to the Noah scheme.